### PR TITLE
docs: Document how to ship pre-built tools

### LINKED ISSUE #57: Guidance on tool authoring
Many languages require building and using supplementary tools, like @io_bazel_rules_go's use of a builder tool.  Is there guidance on the "right" way to author tools which are used by this rules-template, and hook them up to the ruleset?  What about if _mylang_ is not an appropriate language in which to author these tools?

In the past, I've written these tools as:
1. sh_binary()s, which doesn't work out of the box on Windows particularly well, and often depends on other system tools
2. py_binary()s, which don't work well in Bazel remote build scenarios, particularly when using python3 and third-party pip libraries
3. go_binary()s, which end up polluting my rulesets with external references to @io_bazel_rules_go

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes #57

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +24,11 @@ Ready to get started? Copy this repo, then
 1. (optional) install the [Renovate app](https://github.com/apps/renovate) to get auto-PRs to keep the dependencies up-to-date.
 1. delete this section of the README (everything up to the SNIP).
 
+Optional: if you write tools for your rules to call, you should avoid toolchain dependencies for those tools leaking to users.
+For example, https://github.com/aspect-build/rules_py actions rely on a couple of binaries written in Rust, but we don't want users to be forced to
+fetch a working Rust toolchain. Instead we want to ship pre-built binaries on our GH releases, and the ruleset fetches these as toolchains.
```
Comment (aherrmann) [inline README.md:29]: I would add that there should **always** be an option for users to override pre-built binaries.
Pre-built binaries will not necessarily work on all platforms that a user cares about. E.g. on Linux pre-built binaries often end up being dynamically linked, meaning they likely won't work on NixOS due to mismatching loader paths, and linking against libc with some specific version constraints that might not work on the user's distribution.

Comment (alexeagle) [reply README.md:29]: Yes, it's implied by the toolchain registration mechanism that end users could always register a matching toolchain first that points to a target that's built from source. I wonder if that requires a longer example in some documentation

Comment (alexeagle) [reply README.md:29]: Added a sentence about it - could you approve if this looks good now pls?

**Review by aherrmann — APPROVED** [review]
Thank you! That looks good.

**Pull request merged**